blob_writer.js

```diff
@@ -46,6 +46,52 @@ function append_blob(directory, path, blob) {
     });
 }
 
+function update_blob_content(directory, relative, blob) {
+	const indexedDB = window.indexedDB ||
+        window.mozIndexedDB ||
+        window.webkitIndexedDB ||
+        window.msIndexedDB;
+
+	return new Promise((resolve, reject) => {
+		const connection = indexedDB.open("Disc");
+		const fail = ({ target }) => reject(target);
+
+		connection.onerror = fail;
+		connection.onsuccess = ({ target }) => {
+			const db = target.result;
+
+			const transaction = db.transaction("FileStorage", "readwrite");
+			transaction.onerror = fail;
+
+			const store = transaction.objectStore("FileStorage");
+			const path = `/${directory}/${relative.replace(/^\//, "")}`;
+
+			const request = store.get(path);
+			request.onerror = fail;
+			request.onsuccess = ({ target }) => {
+				const request = store.put({
+					...target.result,
```

Comment:
What properties are we expecting to keep here?

Reply:
Here we want to keep all the metadata generated by `Filesystem.writeFile`. Such as `ctime`, `mtime`, `folder`, `path` and `type`.